pkg/client/group_mgmt_client.go

```diff
@@ -159,6 +190,15 @@ func (client *GroupMgmtClient) Put(path, id string, payload interface{}, respHol
 		if err != nil {
 			return nil, err
 		}
+		// refresh session token
+		isSessionRefresh, err := client.refreshSessionToken(response.StatusCode(), errResp)
+		if err != nil {
+			return nil, err
+		}
+		if isSessionRefresh {
+			// retry the ops
+			goto RETRY
+		}
```

Comment:
goto statements are rarely a good option when you can opt to do something else.

Why is this code copied in every function (Get, Post, Put, etc.)?  Why not simply move this code into a function (e.g. verifyAuth) which checks for the 401 code and attempts to recreate the session?

Regarding the goto, please investigate the RetryCondition option in go-resty.  If that doesn't pan out, I still prefer we find another way to perform the retry rather than a goto.

Reply:
>> Why is this code copied in every function (Get, Post, Put, etc.)? Why not simply move this code into a function (e.g. verifyAuth) which checks for the 401 code and attempts to recreate the session?

We need refresh the token in case of 401 and also we need to retry the failed operation so it has to go in (Get, Post, Put etc ). 

I will check go-resty , other way could be to use for{} loop instead of goto.